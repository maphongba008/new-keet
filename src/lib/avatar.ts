import { ViewStyle } from 'react-native'
import b4a from 'b4a'

import { DEFAULT_AVATAR_COLORS } from './commonStyles'

export const SEED_BITS = 32
export const EMOJI_REGEXP = /\p{Emoji}/u

export const name2seed = (name: string) => {
  if (name.length === SEED_BITS) {
    return name
  } else if (name.length > SEED_BITS) {
    return name.substring(0, SEED_BITS - 1)
  }
  return name.padEnd(SEED_BITS)
}

function randInt(min: number, max: number, seed?: string): number {
  const rand = seed
    ? b4a.readUInt32LE(b4a.from(seed, 'hex')) / (2 ** SEED_BITS - 1)
    : Math.random()
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(rand * (max - min)) + min
}

export const randomColor = (seed?: string) => {
  return DEFAULT_AVATAR_COLORS[randInt(0, DEFAULT_AVATAR_COLORS.length, seed)]
}

export function getScaledFontSize({
  width,
  height,
}: Pick<ViewStyle, 'width' | 'height'>): number {
  const baseWidth = 30
  const baseHeight = 30
  const baseFontSize = 12
  const maxFontThreshold = 40

  if (typeof height !== 'number' || typeof width !== 'number') {
    return baseFontSize
  }

  // Calculate scaling factor
  const scalingFactor = (width * height) / (baseWidth * baseHeight)

  // Scale font size
  const fontSize = baseFontSize * Math.sqrt(scalingFactor)
  const minThreshold = Math.max(fontSize, baseFontSize)

  return Math.min(minThreshold, maxFontThreshold)
}
