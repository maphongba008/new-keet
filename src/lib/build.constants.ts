import Config from 'react-native-config'

const PRODUCT = {
  // define flags in alphabetical order
  FORCE_UPGRADE_BANNER: true,
  GRADIENT_TAB_BAR_ICON: true,
  IS_CHAT_CACHE_ENABLED: true,
  IS_SENTRY_ENABLED: false,
  IS_TIME_STATS_ENABLED: true,
  SHOW_ADMIN_RESET_UI: false,
  SHOW_BROADCAST_FILTER: false,
  SHOW_CHANNEL_UI: true,
  SHOW_CLEAN_DEVICE_SCREEN: false,
  SHOW_FULLSCREEN_AD: false,
  SHOW_MESSAGE_COPY_LINK: false,
  SHOW_NEW_PROFILE_FEATURE: false,
  SHOW_REMOVE_DEVICE: false,
  SHOW_RENAME_DEVICE: false,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PASSCODE: false,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: false,
  SHOW_UPDATE_BANNER: true,
  SHOW_WALLET: false,
  TRANSPARENT_TAB_BAR: true,
}

// KEEP this eslint statement or will raise lint error when swith to other setup
const STAGING = {
  ...PRODUCT,
  // define flags in alphabetical order
  IS_CHAT_CACHE_ENABLED: true,
  IS_SENTRY_ENABLED: false,
  SHOW_BROADCAST_FILTER: false,
  SHOW_CHANNEL_UI: true,
  SHOW_CLEAN_DEVICE_SCREEN: true,
  SHOW_MESSAGE_COPY_LINK: true,
  SHOW_NEW_PROFILE_FEATURE: true,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PASSCODE: true,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: true,
  SHOW_WALLET: true,
  SHOW_ADMIN_RESET_UI: true,
  SHOW_REMOVE_DEVICE: true,
  SHOW_RENAME_DEVICE: true,
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const SV = {
  ...PRODUCT,
  // define flags in alphabetical order
  GRADIENT_TAB_BAR_ICON: false,
  SHOW_PARTICIPANT_SHEET_USER_NAME: false,
  SHOW_PUSH_NOTIFICATION_OPTION: false,
  SHOW_ROOM_FILTER: true,
  SHOW_UPDATE_BANNER: false,
  TRANSPARENT_TAB_BAR: false,
  IS_CHAT_CACHE_ENABLED: true,
}

// Specify a feature set alphabetically
const APP = STAGING

export const FORCE_UPGRADE_BANNER = APP?.FORCE_UPGRADE_BANNER

export const GRADIENT_TAB_BAR_ICON = APP?.GRADIENT_TAB_BAR_ICON

export const IS_SENTRY_ENABLED = APP?.IS_SENTRY_ENABLED

export const SHOW_BROADCAST_FILTER = APP?.SHOW_BROADCAST_FILTER

export const SHOW_CHANNEL_UI = APP?.SHOW_CHANNEL_UI

export const SHOW_CLEAN_DEVICE_SCREEN = APP?.SHOW_CLEAN_DEVICE_SCREEN

export const SHOW_FULLSCREEN_AD = APP?.SHOW_FULLSCREEN_AD

export const SHOW_MESSAGE_COPY_LINK = APP?.SHOW_MESSAGE_COPY_LINK

export const SHOW_NEW_PROFILE_FEATURE = APP?.SHOW_NEW_PROFILE_FEATURE

export const SHOW_PARTICIPANT_SHEET_USER_NAME =
  APP?.SHOW_PARTICIPANT_SHEET_USER_NAME

export const SHOW_PASSCODE = APP?.SHOW_PASSCODE

export const SHOW_PUSH_NOTIFICATION_OPTION = APP?.SHOW_PUSH_NOTIFICATION_OPTION

export const SHOW_ROOM_FILTER = APP?.SHOW_ROOM_FILTER

export const SHOW_UPDATE_BANNER = APP?.SHOW_UPDATE_BANNER

export const SHOW_WALLET = APP?.SHOW_WALLET

export const TRANSPARENT_TAB_BAR = APP?.TRANSPARENT_TAB_BAR

export const IS_EXPERIMENTAL_BUILD = Config?.KEET_VARIANT === 'experimental'

export const IS_CHAT_CACHE_ENABLED = APP?.IS_CHAT_CACHE_ENABLED

export const IS_TIME_STATS_ENABLED = APP?.IS_TIME_STATS_ENABLED

export const SHOW_ADMIN_RESET_UI = APP?.SHOW_ADMIN_RESET_UI

export const SHOW_REMOVE_DEVICE = APP?.SHOW_REMOVE_DEVICE

export const SHOW_RENAME_DEVICE = APP?.SHOW_RENAME_DEVICE
