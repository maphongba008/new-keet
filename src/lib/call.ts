import { Alert } from 'react-native'

import { callLeave } from '@holepunchto/keet-store/store/call'

import type { Strings } from 'i18n/strings'

import { doVibrateSuccess } from './haptics'
import { getDispatch } from './store'

type showCallOnGoingAlertProps = {
  onPressLeave: Function
  onPressStay?: Function
  alertDesc?: string
  strings: Strings
}

export const showCallOnGoingAlert = ({
  onPressLeave,
  onPressStay,
  alertDesc,
  strings,
}: showCallOnGoingAlertProps) => {
  Alert.alert(
    strings.call.alertTitle,
    alertDesc || strings.call.alertDesc,
    [
      {
        text: strings.call.stayInCall,
        onPress: () => {
          onPressStay?.()
        },
        style: 'cancel',
      },
      {
        text: strings.call.leaveCall,
        onPress: () => {
          onPressLeave?.()
        },
        style: 'destructive',
      },
    ],
    { cancelable: true },
  )
}

export const dismissCall = () => {
  try {
    const dispatch = getDispatch()
    dispatch(callLeave())
    doVibrateSuccess()
  } catch (err) {
    console.log('Error in dismissCall', err)
  }
}
