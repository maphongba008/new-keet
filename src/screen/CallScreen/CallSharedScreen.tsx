import React, { memo, useCallback, useEffect, useState } from 'react'
import { StyleSheet, View } from 'react-native'
import isEqual from 'react-fast-compare'
import { Zoom } from 'react-native-reanimated-zoom'

import { HolepunchWebRTCVideoView } from '@holepunchto/webrtc'

import { IconButton } from 'component/Button'
import SvgIcon from 'component/SvgIcon'
import { colors, createThemedStylesheet } from 'component/theme'
import { APPIUM_IDs } from 'lib/appium'
import s, {
  ICON_SIZE_16,
  UI_SIZE_8,
  UI_SIZE_16,
  UI_SIZE_32,
} from 'lib/commonStyles'
import { useMember } from 'lib/hooks/useMember'
import {
  lockScreenPortraitUpAsync,
  unlockScreenAsync,
} from 'lib/screenOrientation'

import CallAllMembersNew from './CallAllMembersNew'
import { CallMemberNameWithIcon } from './CallMemberName'
import { CallSharedScreenProps } from './type'

const CallSharedScreen = ({
  roomId,
  streamTrackId,
  streamMemberId,
}: CallSharedScreenProps) => {
  const { member: sharedScreenMember } = useMember(roomId, streamMemberId)
  const [expanded, setExpanded] = useState(false)

  useEffect(() => {
    return () => {
      lockScreenPortraitUpAsync()
      setExpanded(false)
    }
  }, [setExpanded])

  const toggleExpanded = useCallback(() => {
    setExpanded((_expanded) => {
      if (!_expanded) {
        unlockScreenAsync()
      } else {
        lockScreenPortraitUpAsync()
      }

      return !_expanded
    })
  }, [setExpanded])

  const styles = getStyles()

  if (!streamTrackId) return null

  return (
    <View style={s.container}>
      <View style={styles.sharedScreenContainer}>
        <View style={[s.overflowHidden, styles.sharedScreenRTCContainer]}>
          <Zoom style={s.container}>
            <HolepunchWebRTCVideoView
              trackId={streamTrackId}
              style={styles.sharedScreenRTCContainer}
            />
          </Zoom>
        </View>
        <IconButton
          style={styles.resizeButton}
          onPress={toggleExpanded}
          testID={APPIUM_IDs.call_shared_screen_expand_button}
        >
          <SvgIcon
            color={colors.keet_grey_100}
            name={
              expanded
                ? 'arrowDownLeftAndArrowUpRightToCenter'
                : 'arrowUpRightAndArrowDownLeftFromCenter'
            }
            width={ICON_SIZE_16}
            height={ICON_SIZE_16}
          />
        </IconButton>
        <CallMemberNameWithIcon
          member={sharedScreenMember}
          isMuted={false}
          roomId={roomId}
        />
      </View>
      {!expanded && (
        <CallAllMembersNew roomId={roomId} style={styles.allMembersContainer} />
      )}
    </View>
  )
}

export default memo(CallSharedScreen, isEqual)

const getStyles = createThemedStylesheet((theme) => {
  const styles = StyleSheet.create({
    allMembersContainer: {
      marginTop: UI_SIZE_8,
    },
    resizeButton: {
      backgroundColor: theme.background.bg_2,
      borderRadius: UI_SIZE_8,
      height: undefined,
      padding: UI_SIZE_16,
      position: 'absolute',
      right: UI_SIZE_16,
      top: UI_SIZE_16,
      width: undefined,
    },
    sharedScreenContainer: {
      borderColor: theme.border.color,
      borderRadius: UI_SIZE_32,
      borderWidth: theme.border.width,
      flex: 1,
      marginHorizontal: UI_SIZE_16,
    },
    sharedScreenRTCContainer: {
      borderRadius: UI_SIZE_32,
      flex: 1,
    },
  })
  return styles
})
