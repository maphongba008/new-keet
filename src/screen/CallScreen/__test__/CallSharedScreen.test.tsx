import React from 'react'
import { screen, userEvent } from '@testing-library/react-native'

import { APPIUM_IDs } from 'lib/appium'
import { renderWithProviders } from 'lib/testUtils'

import CallSharedScreen from '../CallSharedScreen'

jest.mock('react-native-reanimated', () => ({
  ...jest.requireActual('react-native-reanimated/mock'),
}))

describe('CallSharedScreen component tests', () => {
  it('renders null with empty stream', () => {
    const element = (
      <CallSharedScreen roomId="" streamTrackId="" streamMemberId="" />
    )
    renderWithProviders(element)
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('renders with valid stream', () => {
    const element = (
      <CallSharedScreen roomId="" streamTrackId="trackId" streamMemberId="" />
    )
    renderWithProviders(element)
    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('renders with valid stream and expand button press', async () => {
    const element = (
      <CallSharedScreen roomId="" streamTrackId="trackId" streamMemberId="" />
    )
    renderWithProviders(element)
    const expandButton = await screen.findByTestId(
      APPIUM_IDs.call_shared_screen_expand_button,
    )
    expect(expandButton).toBeOnTheScreen()
    const user = userEvent.setup()
    await user.press(expandButton)
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
