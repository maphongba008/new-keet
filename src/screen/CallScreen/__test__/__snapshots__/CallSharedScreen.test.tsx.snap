// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallSharedScreen component tests renders null with empty stream 1`] = `null`;

exports[`CallSharedScreen component tests renders with valid stream 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#323A4C",
        "borderRadius": 32,
        "borderWidth": 1,
        "flex": 1,
        "marginHorizontal": 16,
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 32,
            "flex": 1,
          },
        ]
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        style={
          [
            {
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            {
              "borderRadius": 32,
              "flex": 1,
            }
          }
          testID="webrtc_video"
          trackId="trackId"
        >
          <Text>
            HolepunchWebRTCVideoView
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="call_shared_screen_expand_button"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "opacity": 1,
          },
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            {
              "height": 30,
              "width": 30,
            },
            {
              "backgroundColor": "#151823",
              "borderRadius": 8,
              "height": undefined,
              "padding": 16,
              "position": "absolute",
              "right": 16,
              "top": 16,
              "width": undefined,
            },
          ],
        ]
      }
      testID="call_shared_screen_expand_button"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={16}
        bbWidth={16}
        color="#CED3DC"
        fill="#CED3DC"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {},
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        testID="svg-icon-arrowUpRightAndArrowDownLeftFromCenter"
        tintColor="#CED3DC"
        vbHeight={24}
        vbWidth={24}
        width={16}
        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M20.063 4.5V9a.563.563 0 0 1-1.125 0V5.858l-4.29 4.29a.562.562 0 0 1-.796-.796l4.29-4.29H15a.562.562 0 1 1 0-1.125h4.5a.56.56 0 0 1 .563.563m-10.71 9.353-4.29 4.29V15a.562.562 0 1 0-1.125 0v4.5a.56.56 0 0 0 .562.563H9a.563.563 0 0 0 0-1.125H5.858l4.29-4.29a.562.562 0 0 0-.796-.796m10.147.585a.56.56 0 0 0-.562.562v3.142l-4.29-4.29a.562.562 0 0 0-.796.796l4.29 4.29H15a.562.562 0 1 0 0 1.125h4.5a.56.56 0 0 0 .563-.563V15a.56.56 0 0 0-.563-.562M5.858 5.063H9a.563.563 0 1 0 0-1.125H4.5a.563.563 0 0 0-.562.562V9a.563.563 0 1 0 1.124 0V5.858l4.29 4.29a.562.562 0 0 0 .796-.796z" fill="#fff"/></svg>
"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4291744732,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M20.063 4.5V9a.563.563 0 0 1-1.125 0V5.858l-4.29 4.29a.562.562 0 0 1-.796-.796l4.29-4.29H15a.562.562 0 1 1 0-1.125h4.5a.56.56 0 0 1 .563.563m-10.71 9.353-4.29 4.29V15a.562.562 0 1 0-1.125 0v4.5a.56.56 0 0 0 .562.563H9a.563.563 0 0 0 0-1.125H5.858l4.29-4.29a.562.562 0 0 0-.796-.796m10.147.585a.56.56 0 0 0-.562.562v3.142l-4.29-4.29a.562.562 0 0 0-.796.796l4.29 4.29H15a.562.562 0 1 0 0 1.125h4.5a.56.56 0 0 0 .563-.563V15a.56.56 0 0 0-.563-.562M5.858 5.063H9a.563.563 0 1 0 0-1.125H4.5a.563.563 0 0 0-.562.562V9a.563.563 0 1 0 1.124 0V5.858l4.29 4.29a.562.562 0 0 0 .796-.796z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#3E404B",
            "borderRadius": 16,
            "bottom": 8,
            "flexDirection": "row",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 16,
          },
          undefined,
          undefined,
          {
            "maxWidth": undefined,
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={12}
        bbWidth={12}
        color="#26D2E8"
        fill="#26D2E8"
        focusable={false}
        height={12}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {},
            {
              "flex": 0,
              "height": 12,
              "width": 12,
            },
          ]
        }
        testID="svg-icon-waveformLines"
        tintColor="#26D2E8"
        vbHeight={24}
        vbWidth={24}
        width={12}
        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M5.25 9v6a.75.75 0 1 1-1.5 0V9a.75.75 0 0 1 1.5 0m3-6.75A.75.75 0 0 0 7.5 3v18A.75.75 0 1 0 9 21V3a.75.75 0 0 0-.75-.75m3.75 3a.75.75 0 0 0-.75.75v12a.75.75 0 1 0 1.5 0V6a.75.75 0 0 0-.75-.75m3.75 3A.75.75 0 0 0 15 9v6a.75.75 0 1 0 1.5 0V9a.75.75 0 0 0-.75-.75m3.75-1.5a.75.75 0 0 0-.75.75v9a.75.75 0 1 0 1.5 0v-9a.75.75 0 0 0-.75-.75" fill="#fff"/></svg>
"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4280734440,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M5.25 9v6a.75.75 0 1 1-1.5 0V9a.75.75 0 0 1 1.5 0m3-6.75A.75.75 0 0 0 7.5 3v18A.75.75 0 1 0 9 21V3a.75.75 0 0 0-.75-.75m3.75 3a.75.75 0 0 0-.75.75v12a.75.75 0 1 0 1.5 0V6a.75.75 0 0 0-.75-.75m3.75 3A.75.75 0 0 0 15 9v6a.75.75 0 1 0 1.5 0V9a.75.75 0 0 0-.75-.75m3.75-1.5a.75.75 0 0 0-.75.75v9a.75.75 0 1 0 1.5 0v-9a.75.75 0 0 0-.75-.75"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "flexShrink": 1,
              "marginTop": 12,
            },
            {
              "maxWidth": undefined,
            },
            {
              "marginTop": 0,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "color": "#ffffff",
                "flexShrink": 1,
                "fontFamily": "OpenSans-Regular",
                "fontSize": 12,
                "includeFontPadding": false,
                "marginHorizontal": 4,
              },
              {
                "color": "#ffffff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 10,
                "includeFontPadding": false,
                "marginLeft": 4,
              },
              undefined,
            ]
          }
        >
          Connecting...
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flex": 1,
          "paddingTop": 8,
        },
        {
          "marginTop": 8,
        },
      ]
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`CallSharedScreen component tests renders with valid stream and expand button press 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#323A4C",
        "borderRadius": 32,
        "borderWidth": 1,
        "flex": 1,
        "marginHorizontal": 16,
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 32,
            "flex": 1,
          },
        ]
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        style={
          [
            {
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
                {
                  "scale": 1,
                },
              ],
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            {
              "borderRadius": 32,
              "flex": 1,
            }
          }
          testID="webrtc_video"
          trackId="trackId"
        >
          <Text>
            HolepunchWebRTCVideoView
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="call_shared_screen_expand_button"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "opacity": 0.66,
          },
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            {
              "height": 30,
              "width": 30,
            },
            {
              "backgroundColor": "#151823",
              "borderRadius": 8,
              "height": undefined,
              "padding": 16,
              "position": "absolute",
              "right": 16,
              "top": 16,
              "width": undefined,
            },
          ],
        ]
      }
      testID="call_shared_screen_expand_button"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={16}
        bbWidth={16}
        color="#CED3DC"
        fill="#CED3DC"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {},
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        testID="svg-icon-arrowDownLeftAndArrowUpRightToCenter"
        tintColor="#CED3DC"
        vbHeight={24}
        vbWidth={24}
        width={16}
        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="m19.898 4.898-5.04 5.04H18a.562.562 0 1 1 0 1.124h-4.5a.563.563 0 0 1-.562-.562V6a.563.563 0 0 1 1.124 0v3.143l5.04-5.04a.562.562 0 0 1 .796.795m-9.398 8.04H6a.562.562 0 0 0 0 1.124h3.142l-5.04 5.04a.563.563 0 1 0 .795.796l5.04-5.04V18a.563.563 0 0 0 1.126 0v-4.5a.56.56 0 0 0-.563-.562" fill="#fff"/></svg>
"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4291744732,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m19.898 4.898-5.04 5.04H18a.562.562 0 1 1 0 1.124h-4.5a.563.563 0 0 1-.562-.562V6a.563.563 0 0 1 1.124 0v3.143l5.04-5.04a.562.562 0 0 1 .796.795m-9.398 8.04H6a.562.562 0 0 0 0 1.124h3.142l-5.04 5.04a.563.563 0 1 0 .795.796l5.04-5.04V18a.563.563 0 0 0 1.126 0v-4.5a.56.56 0 0 0-.563-.562"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#3E404B",
            "borderRadius": 16,
            "bottom": 8,
            "flexDirection": "row",
            "paddingHorizontal": 8,
            "paddingVertical": 4,
            "position": "absolute",
            "right": 16,
          },
          undefined,
          undefined,
          {
            "maxWidth": undefined,
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={12}
        bbWidth={12}
        color="#26D2E8"
        fill="#26D2E8"
        focusable={false}
        height={12}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {},
            {
              "flex": 0,
              "height": 12,
              "width": 12,
            },
          ]
        }
        testID="svg-icon-waveformLines"
        tintColor="#26D2E8"
        vbHeight={24}
        vbWidth={24}
        width={12}
        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M5.25 9v6a.75.75 0 1 1-1.5 0V9a.75.75 0 0 1 1.5 0m3-6.75A.75.75 0 0 0 7.5 3v18A.75.75 0 1 0 9 21V3a.75.75 0 0 0-.75-.75m3.75 3a.75.75 0 0 0-.75.75v12a.75.75 0 1 0 1.5 0V6a.75.75 0 0 0-.75-.75m3.75 3A.75.75 0 0 0 15 9v6a.75.75 0 1 0 1.5 0V9a.75.75 0 0 0-.75-.75m3.75-1.5a.75.75 0 0 0-.75.75v9a.75.75 0 1 0 1.5 0v-9a.75.75 0 0 0-.75-.75" fill="#fff"/></svg>
"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4280734440,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M5.25 9v6a.75.75 0 1 1-1.5 0V9a.75.75 0 0 1 1.5 0m3-6.75A.75.75 0 0 0 7.5 3v18A.75.75 0 1 0 9 21V3a.75.75 0 0 0-.75-.75m3.75 3a.75.75 0 0 0-.75.75v12a.75.75 0 1 0 1.5 0V6a.75.75 0 0 0-.75-.75m3.75 3A.75.75 0 0 0 15 9v6a.75.75 0 1 0 1.5 0V9a.75.75 0 0 0-.75-.75m3.75-1.5a.75.75 0 0 0-.75.75v9a.75.75 0 1 0 1.5 0v-9a.75.75 0 0 0-.75-.75"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "flexShrink": 1,
              "marginTop": 12,
            },
            {
              "maxWidth": undefined,
            },
            {
              "marginTop": 0,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "color": "#ffffff",
                "flexShrink": 1,
                "fontFamily": "OpenSans-Regular",
                "fontSize": 12,
                "includeFontPadding": false,
                "marginHorizontal": 4,
              },
              {
                "color": "#ffffff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 10,
                "includeFontPadding": false,
                "marginLeft": 4,
              },
              undefined,
            ]
          }
        >
          Connecting...
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
